<launch>
  <!-- image_proc를 /usb_cam 네임스페이스에 배치 -->
  <group ns="usb_cam">
    <node name="image_proc" pkg="image_proc" type="image_proc" output="screen"/>
    
    <!--불필요한 퍼블리셔 플러그인 비활성화 -->
    <rosparam param="disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
    
    <rosparam param="image_raw/disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>

    <rosparam param="image_rect/disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
    <rosparam param="image_rect_color/disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>

    <rosparam param="image_color/disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'

    </rosparam>
        <rosparam param="image_mono/disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
  </group>  

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="/dev/video0"/>
    <param name="image_width" value="640"/>
    <param name="image_height" value="480"/>
    <param name="pixel_format" value="mjpeg"/>
    <param name="camera_frame_id" value="usb_cam"/>
    <param name="io_method" value="mmap"/>
    <!-- calib YAML -->
    <!-- <param name="camera_info_url" value="file://$(find usb_cam)/calibration/cam_calibration.yaml"/> -->
    <param name="camera_info_url" value="file://$(find autorace_perception)/config/cam_calibration.yaml"/>
  </node>

  <!-- 시각화 
  <node name="image_view" pkg="image_view" type="image_view" output="screen">
    <remap from="image" to="/usb_cam/image_rect_color"/>
    <param name="autosize" value="true"/>
  </node> -->
</launch>

<!--roslaunch autorace_perception usb_cam.launch-->
<!--rqt_image_view /usb_cam/image_rect_color-->
